### DISCUSSION TITLE
LazyVim bindings for VSCode

### DISCUSSION THREAD

**Discussion by brotherkaif** [LazyVim/LazyVim#1443] [Show and tell]
Hello! Not sure if anyone has any interest in this, but I thought I'd share anyway.

Over the past few weeks I've been mapping the LazyVim keybindings _to_ VSCode (that's right, setting up VSCode to behave like LazyVim).

Why am I doing this? LazyVim is my daily driver, but sometimes, I'm _forced_ to use VSCode for liveshare pairing sessions. But, I'm so used to the LazyVim keybindings that I find myself fumbling around VSCode whenever I have to use it.

This `settings.json` file for VSCode is my attempt to map the same LazyVim bindings to the equivalent VSCode functionality. It's certainly not going to be _exactly_ like LazyVim, but at least the same functionality appears when you use the same keybindings (e.g. `<leader>gg` will bring up the source control tool in VSCode instead of LazyGit in LazyVim).

If you're interested, here's [a gist with the config](https://gist.github.com/brotherkaif/098f14950d538cb7001a7e04d8d884ee).

A couple of things to note:
- You need to have the [VSCode Vim](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim) plugin installed for this to work
- I've laid out the config to be identical to the [LazyVim Keymap documentation page](https://www.lazyvim.org/keymaps) to make it easier to read
- There is a lot of mappings that are commented out, either because there isn't a suitable VSCode equivalent, or I just haven't found one yet

Like I said, not sure if anyone has any interest in this, but if anyone has any suggestions for some of the missing ones I'd be more than happy to update the gist.

Thanks!

**Comment by spector700** [comment]
I am trying to do the same thing, except with the vscode Neovim extension.

Is there a reason that you chose the regular vim one?

**Reply by brotherkaif** [reply]
I've only ever had experience of using the VSCode Vim extension, so I went with that one.

From what I understand, the Neovim plugin requires Neovim to be installed as a dependency (as opposed to the VSCode Vim extension which is self contained).

I sometimes join liveshare coding sessions using the web-based VSCode interface which doesn't have access to an underlying devcontainer instance, meaning that it wouldn't be possible to install the required Neovim binary which might make it a bit tricky for me to use that extension unfortunately.

There is the option to use GitHub code spaces instead which does allow you to configure a devcontainer and install Neovim underneath, but I'd need to spend a bit of time looking into it.

Out of interest, how are the keymaps configured in the Neovim extension? I'd be keen to give it a go.

**Reply by cStralpt** [reply]
i've tried those two extensions before, but i prefer to use VSCode Neovim because it's more light weight, this extension actually connect to your lazyvim instead of emulating it, so you need to really configure your config to work just for vs code because in the background when we press leader-gg it actually opened lazygit, we need to be carefully otherwise we will get lost inside of that plugin.
i also have a lazyvim config for this, in order to work with vs code, now i don't need to memorize all of those shortcuts for the both editor.
[https://github.com/cStralpt/lazycodium-starter-template.git](url)

**Reply by brotherkaif** [reply]
Thanks for sharing @cStralpt. I would love to take a look at the config you provided, although unfortunately I think it's private so I can't open the link.

**Reply by cStralpt** [reply]
i just updated the link

**Reply by spector700** [reply]
Thanks, that is what I was looking for,
So why do you have backspace passed?
  {
    "command": "vscode-neovim.send",
    "key": "backspace",
    "when": "editorTextFocus && neovim.mode == insert",
    "args": "<BS>"
  },

**Reply by spector700** [reply]
@cStralpt also, the main issue I always have is with mini.surround. Most of the time it doesn't work.
Have you added binds to your VS Code to make this kind of stuff work better?

**Reply by cStralpt** [reply]
@spector700 that keybinding is just like `<C-w>` but backspace is doing the opposite, and the mini.surround is remapped with gsa(z->s) it working fine on my PC, both on Ubuntu and Windows 11.

**Reply by spector700** [reply]
Yeah I remapped my surround to the same and I works fine in nvim but vscode just doesn't like it most of the time.

**Reply by cStralpt** [reply]
maybe you dont configure the extension properly, there's a feature that tell you, weather your vs code get connected or not into lazyvim, you can look at the status bar:
![image](https://github.com/LazyVim/LazyVim/assets/95400822/60b6050a-bd0b-4fa8-9bce-611a833aac70)
if that does not appear than you're not actually connected to lazyvim, you need to check your VS Code Neovim setup.

**Reply by spector700** [reply]
Yeah, I already stole a lot of your config, but it connects fine.
Do you have a keybindings.json that is not in your repo?

**Reply by cStralpt** [reply]
i think you need to add gsa key to your keybindings in order to make it work, because i used to do the same thing to fix problems like this on my config.

**Reply by cgjosephlee** [reply]
I have this in my config and `mini.surround` works nice in VSCode, hope it helps.
https://github.com/cgjosephlee/dotfiles/blob/master/.config/nvim/lua/plugins/vscode.lua

**Reply by brendancopley** [reply]
@cgjosephlee I fixed the link for ya
https://github.com/cgjosephlee/dotfiles/blob/v20231114/.config/nvim/lua/plugins/vscode.lua
